source/Calamari.Common/Features/Packages/Decorators/PackageEntryExtractorDecorator.cs

```diff
@@ -0,0 +1,33 @@
+﻿using SharpCompress.Archives;
+
+namespace Calamari.Common.Features.Packages.Decorators
+{
+    /// <summary>
+    /// A base Decorator which allows addition to the behaviour of an IPackageEntryExtractor 
+    /// </summary>
+    /// <remarks>
+    /// IPackageEntryExtractor is a more specific IPackageExtractor, which provides a hook to the extraction
+    /// of each individual ArchiveEntry.
+    /// </remarks>
+    public class PackageEntryExtractorDecorator : IPackageEntryExtractor
```

Comment:
Whoops, this is currently unused as I removed the only decorator that was deriving from it. The functionality and abstractions needed a rethink.

Reply:
This needs to be removed, but I'll do in a follow up PR.